pyaptamer/datasets/_loaders/load_aptamer_interactions.py

```diff
@@ -0,0 +1,235 @@
+__author__ = "Satarupa22-SD"
+__all__ = ["load_aptadb", "load_aptamer_interactions", "load_interactions"]
+
+from pathlib import Path
+
+import pandas as pd
+
+
+def _download_dataset(
+    dataset_name: str, target_dir: Path, force_download: bool = False
+) -> None:
+    """Download a Kaggle dataset to the specified directory and unzip it.
+
+    This is a private helper function used internally by the module.
+
+    Parameters
+    ----------
+    dataset_name : str
+        The Kaggle dataset identifier in format "username/dataset-name"
+    target_dir : Path
+        Directory where the dataset should be downloaded and extracted
+    force_download : bool, default False
+        If True, download even if CSV files already exist in target_dir
+
+    Raises
+    ------
+    ImportError
+        If the kaggle package is not installed
+    Exception
+        If the download fails for any reason
+
+    Notes
```

Comment:
GPT tends to generate notes, I dont think we should add aditional text to reaad in a new subsection, but it could be developer preference. what do you feel @fkiraly ?

Reply:
I would move it up to the top